StkAutomation/Matlab/STK Object Model/Radar_Scheduler_Deck_Access.m

```diff
@@ -0,0 +1,163 @@
+% %Written By: Alexander Ridgeway
+% %Date:       27 April 2016
+% 
+clear;
+clc;
+try
+    % Grab an existing instance of STK
+    uiapp = actxGetRunningServer('STK12.application');
+catch
+    % STK is not running, launch new instance
+    % Launch a new instance of STK and grab it
+    uiapp = actxserver('STK12.application');
+end
+
+%get the root from the personality
+root = uiapp.Personality2;
+% set visible to true (show STK GUI)
+uiapp.visible = 1;
+%%From the STK Object Root you can command every aspect of the STK GUI
+%close current scenario or open new one
+try
+    root.CloseScenario();
+    root.NewScenario('DeckAccess');
+catch
+    root.NewScenario('DeckAccess');
+end
+%get the scenario root, its of type IAgScenario 
+scen = root.CurrentScenario;
+%root.ExecuteCommand('SetUnits / Km Epsec');
+
+%Create the Radar Site (Delete if it exists)
+try
+    fac = root.CurrentScenario.Children.New('eFacility','Radar');
+catch
+    fac = root.GetObjectFromPath('/Facility/Radar');
+    fac.Unload;
+    fac = root.CurrentScenario.Children.New('eFacility','Radar');
+end
+
+FacLat = 13.7355;
+FacLon = 80.1847;
+FacAlt = -0.0564551;
+RangeMax = 1000;
+Start = '1 Jan 2017 00:00:00.000';
+Stop = '1 Jan 2017 00:30:00.000';
+
+
+fac.UseTerrain = false;
+fac.Position.AssignGeodetic(FacLat,FacLon,FacAlt);
+sen = fac.Children.New('eSensor','Dome');
+sen.CommonTasks.SetPatternRectangular(65,65);
+const = sen.AccessConstraints;
+range = const.AddConstraint(34); 
+range.EnableMax = true;
+range.Max = RangeMax;
+ 
+Startdate = datetime(Start,'InputFormat','dd MMM yyyy HH:mm:SS.000');
+Stopdate  = datetime(Stop,'InputFormat','dd MMM yyyy HH:mm:SS.000');
+Duration = seconds(Stopdate-Startdate);
+
+disp(['Facility Location'])
+disp(['Latitude:  ',num2str(FacLat),' degrees'])
+disp(['Longitude: ',num2str(FacLon),' degrees'])
+disp(['Altitude:  ',num2str(FacAlt),' degrees'])
+disp([' '])
+disp(['Time of Analysis'])
+disp(['Start:     ',Start,' UTC'])
+disp(['Stop:      ',Stop, ' UTC'])
+disp(['Duration:  ',num2str(Duration),' seconds'])
+disp([' '])
+
+disp('***Begin Deck Access***')
+disp(datetime('now'))
+cmd =['DeckAccess */Facility/Radar/Sensor/Dome "' Start '" "' Stop '" Satellite "C:\ProgramData\AGI\STK 11\Databases\Satellite\stkAllTLE.tce" SortObj OutReport AddSatellites 10000'];
+root.ExecuteCommand(cmd);
+
+disp('***Complete Deck Access ***')
+disp(datetime('now'))
+
+chain = scen.Children.New('eChain','DeckAccess');
+try 
```

Comment:
Same comment here that the Contains function should be used instead of a try-catch statement